linode/manager
# fix: [M3-8538] - Incorrect Commands in Linode CLI Modal

## Description 📝
This PR addresses the issue within the LinodeCLI modal which errornously displayed the `create` command when a user was going through the `clone` workflow. A subsequent issue was discovered impacting the `cURL` utility. In this case the incorrect url path was being used and a source Linode ID was not being provided.

## Changes  🔄
List any change(s) relevant to the reviewer.
- Updates to the `generateCLICommand()` utility function to generate the expected commands based on `create` or `clone` workflows.
- Updates to the `generateCurlCommand()` utility function to generate the correct url path which will be used to send the payload.

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![clone-cli-prod](https://github.com/user-attachments/assets/ea06918b-08aa-4d66-825b-238274fe79e8) | ![clone-cli-pr-branch](https://github.com/user-attachments/assets/a85a74a4-ffcb-4ac7-8c0b-0187a2c35056) |
| ![clone-curl-prod](https://github.com/user-attachments/assets/c4de34be-4aa9-455b-ad3e-0d612a754938) | ![clone-curl-branch-pr](https://github.com/user-attachments/assets/7c55aeff-b897-4455-920a-e97494ac59c5) |


## How to test 🧪

### Prerequisites
(How to setup test environment)
- Have the [LinodeCLI](https://techdocs.akamai.com/cloud-computing/docs/install-and-configure-the-cli) tool configured locally.
- Have at least one Linode instance on the account.

### Reproduction steps
(How to reproduce the issue, if applicable)
- In production, visit the page to create a new Linode instance.
  - Fill out the form fields to create a Linode.
  - Scroll to the bottom of the page and click on the `View Code Snippets` button.
  - Inspect the commands generated for the `Linode CLI` and `cURL` tabs.
- Next, click on the `Clone Linode` tab
  - Select the Linode instance you'd like to clone.
  - Scroll to the bottom of the page and click on the `View Code Snippets` button.
  - Inspect the commands generated for the `Linode CLI` and `cURL` tabs.

### Verification steps
(How to verify changes)
- Check out this branch and go through the same steps as above.
- Inspect the respective commands generated and presented to the user in the LinodeCLI modal for the `create` and `clone` workflows.
- Verify that the changes match the images above.
- Paste the commands onto your terminal running the LinodeCLI and verify that all workflows behave as expected. 

## As an Author, I have considered 🤔
- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support
